core/src/edu/rice/rbox/Replication/ReplicaManagerGrpc.java

```diff
@@ -156,17 +156,46 @@ public void handleUnsubscribe(Rbox.UnsubscribeRequest request, StreamObserver<Em
 
         @Override
         public void handleUpdate(Rbox.UpdateMessage request, StreamObserver<Empty> responseObserver) {
-            // TODO: Call ChangeReceiver
+            logger.log(Level.INFO, "Handling update...");
+
+            // Call ChangeReceiver
+            // TODO: Cannot reference receiveChange from static context?
+            ChangeReceiver.receiveChange(request.getRemoteChange());
+
+            sendToReplicaHolders(request.getTargetObject(), request.getRemoteChange());
```

Comment:
When we are handling update, we don't send to replica holders because the update might be rejected by object storage